src/orquestra/sdk/_base/_storage/orqdantic.py

```diff
@@ -0,0 +1,123 @@
+################################################################################
+# © Copyright 2024 Zapata Computing Inc.
+################################################################################
+
+"""Compatibility layer for pydantic v1 / v2 compatibility."""
+
+from copy import deepcopy
+from typing import Annotated, Any, Optional
+
+import pydantic
+
+PYDANTICV1 = pydantic.__version__.startswith("1.")
+
+if PYDANTICV1:
+    from pydantic.generics import GenericModel
+
+    class BaseModel(GenericModel):
+        @classmethod
+        def model_validate(cls, *args, **kwargs):
+            return super(BaseModel, cls).parse_obj(*args, **kwargs)
+
+        @classmethod
+        def model_validate_json(cls, *args, **kwargs):
+            return super(BaseModel, cls).parse_raw(*args, **kwargs)
+
+        def model_dump(self, *args, **kwargs):
+            return super().dict(*args, **kwargs)
+
+        def model_dump_json(self, *args, **kwargs):
+            return super().json(*args, **kwargs)
+
+        @classmethod
+        def model_json_schema(cls, *args, **kwargs):
+            return super(BaseModel, cls).schema_json(*args, **kwargs)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Any reason these aren't `super(GenericModel, cls) ...`?

Reply:
Nope, [f963f60](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/373/commits/f963f609a521ae890f1e3c69f23530c7ae0fabcd)